### DISCUSSION TITLE
Attach physics bodies to primitives?

### DISCUSSION THREAD

**Discussion by AlexandrosGounis** [pmndrs/use-cannon#111] [General]
Hello, I have a `gltf` model loaded like this:
```ts
 const [model, setModel] = useState()
  useEffect(() => {
    new GLTFLoader().load('aCylinder.gltf', setModel)
  }, [])
```

and then a cylinder...
```ts
const [ref, api] = useCylinder(() => ({
    mass: 3.3,
    args: [0.8, 0.8, 7.2, 16],
    position: [2, 3.8, 1]
  }))

return scene ? <mesh ref={ref}>
    <primitive object={model.scene} position={[2, 3.8, 1]} rotation={[0, 1, 0]} />
  </mesh> : null
```

The model is actually an ancient column and the physics body attached to it is just a cylinder, for simplicity.
It looks like there is a cylinder in the space, but when it is moved (due to some collision), the `gltf` is not updated. So the physics body is moved but it's not attached to the gltf model that remains in place.

**Comment by stockhuman** [comment]
Hey @AlexandrosGounis, I imagine you've figured this out by now, but yes, that will work. See [this demo](https://codesandbox.io/s/cannon-gltf-rodtp) with updated (v6) syntax.
